src/Components/ArticlePreviewSection/ArticlePreviewSection.tsx

```diff
@@ -1,57 +1,25 @@
+"use client";
+
 import ArticleCard from "@/Components/ArticleCard/ArticleCard";
 import styles from "@/Components/ArticlePreviewSection/ArticlePreviewSection.module.scss";
-import { fetchPosts } from "@/lib/api/postsFetcher";
+import { useUserContext } from "@/context/UserPrefsContext";
+import { PostType } from "@/lib/types/postsType";
 
-export type ArticleCardPropsType = {
-  featuredImage: {
-    node: {
-      altText: string;
-      mediaItemUrl: string;
-      mediaDetails: {
-        sizes: Array<{
-          sourceUrl: string;
-          height: number;
-          width: number;
-        }>;
-      };
-    };
-  };
-  id: string;
-  title: string;
-  excerpt: string;
-  content: string;
-  slug: string;
-  tags: {
-    nodes: Array<{ name: string }>;
-  };
-  author: {
-    node: {
-      firstName: string;
-      lastName: string;
-      slug: string;
-    };
-  };
-  date: string;
-  modified: string;
-  categories: {
-    nodes: Array<{ name: string }>;
-  };
-};
+const ArticlePreviewSection = ({ posts }: { posts: PostType[] }) => {
+  const { layoutStyle } = useUserContext();
 
-const ArticlePreviewSection = async ({
-  category,
-  tag,
-}: {
-  category: string;
-  tag: string;
-}) => {
-  const articlesData = await fetchPosts(category, tag);
   return (
-    <div className={styles.articleGridContainer}>
-      <ul className={styles.articleList}>
-        {articlesData ? (
-          articlesData.map((articleData: ArticleCardPropsType) => (
-            <ArticleCard key={articleData.id} postData={articleData} />
+    <div className={styles.articleContainer}>
+      <ul
+        className={`${layoutStyle === "grid" ? styles.articleGrid : styles.articleList}`}
+      >
+        {posts ? (
+          posts.map((articleData: PostType) => (
+            <ArticleCard
+              key={articleData.id}
+              postData={articleData}
+              layoutStyle={layoutStyle}
+            />
           ))
         ) : (
           <span>Loading Articles...</span>
```

Comment:
Still required?

Reply:
Only for testing during development. Will be removed in final build.